trustmanager/keyfilestore.go

```diff
@@ -11,20 +11,27 @@ const (
 	keyExtension = "key"
 )
 
+// PassphraseRetriever is a callback function that should retrieve a passphrase
+// for a given named key. If it should be treated as new passphrase (e.g. with
+// confirmation), createNew will be true. Attempts is passed in so that implementers
+// decide how many chances to give to a human, for example.
+type PassphraseRetriever func(keyName string, createNew bool, attempts int) (passphrase string, giveup bool, err error)
+
 // KeyFileStore persists and manages private keys on disk
 type KeyFileStore struct {
 	FileStore
+	PassphraseRetriever
 }
```

Comment:
I didn't realize you could embed a func in a struct like this. It may be a little clearer to give it an explicit name.